src/main/java/org/fineract/messagegateway/zeebe/ZeebeWorkers.java

```diff
@@ -0,0 +1,211 @@
+                                        zeebeClient.newPublishMessageCommand()
+                                                .messageName(CALLBACK_MESSAGE)
+                                                .correlationKey(id)
+                                                .variables(newVariables)
+                                                .timeToLive(Duration.ofMillis(timeToLive))
+                                                .send()
+                                                .join();
+                                    } else if (exchange.getProperty(MESSAGE_DELIVERY_STATUS).equals(false)) {
+                                        logger.info("Publishing variables: " + newVariables);
+                                        newVariables.put(DELIVERY_ERROR_MESSAGE, exchange.getProperty(DELIVERY_ERROR_INFORMATION));
+                                        newVariables.put(MESSAGE_DELIVERY_STATUS, exchange.getProperty(MESSAGE_DELIVERY_STATUS));
+                                        zeebeClient.newPublishMessageCommand()
+                                                .messageName(CALLBACK_MESSAGE)
+                                                .correlationKey(id)
+                                                .variables(newVariables)
+                                                .timeToLive(Duration.ofMillis(timeToLive))
+                                                .send()
+                                                .join();
+                                    }
+                                } else {
+                                    logger.info("Publishing created variables to variables: " + newVariables);
+                                    newVariables.put(CALLBACK_RETRY_COUNT, exchange.getProperty(RETRY_COUNT_CALLBACK));
+                                    zeebeClient.newSetVariablesCommand(Long.parseLong(exchange.getProperty(INTERNAL_ID).toString()))
+                                            .variables(newVariables)
+                                            .send()
+                                            .join();
+                                }
+                            }
+                        }
+                    }else{
+                        logger.info("Callback Retry Over");
+                        exchange.setProperty(MESSAGE_DELIVERY_STATUS,false);
+                        String id = exchange.getProperty(CORRELATION_ID, String.class);
+                        Map<String, Object> newVariables = new HashMap<>();
+                        newVariables.put(MESSAGE_DELIVERY_STATUS, exchange.getProperty(MESSAGE_DELIVERY_STATUS));
+                        newVariables.put(CALLBACK_RETRY_COUNT,exchange.getProperty(RETRY_COUNT_CALLBACK));
+                        zeebeClient.newSetVariablesCommand(Long.parseLong(exchange.getProperty(INTERNAL_ID).toString()))
```

Comment:
what is internal ID, is it Long always?

Reply:
it need not be long , the process instance key can be longer than Long's range

Reply:
It's a zeebeclient method and has a long type in its signature. So I have to pass it as long.